### PR TITLE
Fix error when loading custom pricing configuration file 

### PR DESCRIPTION
**Pull request by mogaal**
This fixes the bug I reported in https://github.com/opencost/opencost-helm-chart/issues/205, where the `CONFIG_PATH` directory is mounted as a file instead of as a directory:

```
~ $ ls /tmp/custom-config -l
-rw-r--r--    1 root     root           299 May 15 10:27 /tmp/custom-config
~ $ cat /tmp/custom-config 
{
  "CPU": "1.25",
  "GPU": "0.95",
  "RAM": "0.5",
  "description": "Modified pricing configuration.",
  "internetNetworkEgress": "0.12",
  "regionNetworkEgress": "0.01",
  "spotCPU": "0.006655",
  "spotRAM": "0.000892",
  "storage": "0.25",
  "zoneNetworkEgress": "0.01",
  "provider" : "custom"
}~ $
```

This causes the pod to fail with:

```
opencost 2024-05-15T10:27:58.127897889Z ERR error accessing cloud provider configuration: stat /tmp/custom-config/default.json: not a directory
```

### CODE REVIEW

**Comment by mogaal** [comment]
@mattray Hi, when you have some time, pls let me know if this looks good to you 🙏

File: charts/opencost/templates/deployment.yaml [hunk 1]
```diff
@@ -243,7 +243,7 @@ spec:
               readOnly: false
             {{- end }}
             {{- if .Values.opencost.customPricing.enabled }}
-            - mountPath: {{ .Values.opencost.customPricing.configPath }}
+            - mountPath: {{ .Values.opencost.customPricing.configPath }}/{{ include "opencost.configFileName" . }}.json
               name: custom-configs
               subPath: {{ include "opencost.configFileName" . }}.json
```
Comment (mattray) [inline charts/opencost/templates/deployment.yaml:248]: the subPath isn't working?

Comment (mogaal) [reply charts/opencost/templates/deployment.yaml:248]: No, it gets mounted as a file:

```
~ $ ls /tmp/custom-config -l
-rw-r--r--    1 root     root           299 May 15 10:27 /tmp/custom-config
~ $ cat /tmp/custom-config 
{
  "CPU": "1.25",
  "GPU": "0.95",
  "RAM": "0.5",
  "description": "Modified pricing configuration.",
  "internetNetworkEgress": "0.12",
  "regionNetworkEgress": "0.01",
  "spotCPU": "0.006655",
  "spotRAM": "0.000892",
  "storage": "0.25",
  "zoneNetworkEgress": "0.01",
  "provider" : "custom"
}~ $
```

Comment (Riyaz-Behrad) [reply charts/opencost/templates/deployment.yaml:248]: As @mogaal said,
The subPath specifies what key should be loaded from the configmap. The mountPath is the full path to the file loaded, so it should also contain the file's name or else the custom-config would be considered as the mounted file's name. Final applied deployment should be like this:
```
        - mountPath: /tmp/custom-config/default.json
          name: custom-configs
          readOnly: true
          subPath: default.json
```

**Comment by valdisrigdon** [comment]
https://github.com/opencost/opencost-helm-chart/pull/208 should have the full fix for this, as you need to fix the custom metrics too.

**Review by mattray — APPROVED** [review]

**Pull request merged**